openmc/mgxs/library.py

```diff
@@ -1092,60 +1155,121 @@ def create_mg_mode(self, xsdata_names=None, xs_ids=None,
 
         if xsdata_names is not None:
             cv.check_iterable_type('xsdata_names', xsdata_names, basestring)
+        xs_type = 'macro'
+
+        # Initialize MGXS File
+        mgxs_file = openmc.MGXSLibrary(self.energy_groups)
+
+        # Set ID names
         if xs_ids is not None:
             if isinstance(xs_ids, basestring):
                 # If we only have a string lets convert it now to a list
                 # of strings.
-                xs_ids = [xs_ids for i in range(len(self.domains))]
+                xs_id_input = xs_ids
+                xs_ids = []
+                for domain in self.domains:
+                    if self.domain_type is 'mesh':
+                        for subdomain in domain.cell_generator():
+                            xs_ids.append(xs_id_input)
+                    else:
+                        xs_ids.append(xs_id_input)
             else:
                 cv.check_iterable_type('xs_ids', xs_ids, basestring)
         else:
-            xs_ids = ['1m' for i in range(len(self.domains))]
-        xs_type = 'macro'
+            xs_ids = []
+            for domain in self.domains:
+                if self.domain_type is 'mesh':
+                    for subdomain in domain.cell_generator():
+                        xs_ids.append('1m')
+                else:
+                    xs_ids.append('1m')
+
+        if self.domain_type == 'mesh':
+            # Before continuing, we cant build a model automatically if
```

Comment:
I'm not sure I understand what this comment actually means. Can you replace "So check and complain as needed" with something more informative - i.e., what goes in and what comes out?